### PR TITLE
feat: Download runner release via latest release API

### PR DESCRIPTION
**Pull request by npalm**
This PR address the potential problems for downloading the GitHub runner agent, see #2453 

- Download release via latest release API instead of listing release
- Drop support for draf / pre-release, not used for action runners

### CODE REVIEW

**Comment by npalm** [comment]
fix: #2453

File: examples/default/main.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 locals {
-  environment = "default"
+  environment = "niek"
```
Comment (ScottGuymer) [inline examples/default/main.tf:2]: For testing?

File: examples/default/main.tf [hunk 2]
```diff
@@ -41,11 +41,11 @@ module "runners" {
...
   # Grab zip files via lambda_download
-  webhook_lambda_zip                = "lambdas-download/webhook.zip"
-  runner_binaries_syncer_lambda_zip = "lambdas-download/runner-binaries-syncer.zip"
-  runners_lambda_zip                = "lambdas-download/runners.zip"
+  # webhook_lambda_zip                = "lambdas-download/webhook.zip"
+  # runner_binaries_syncer_lambda_zip = "lambdas-download/runner-binaries-syncer.zip"
+  # runners_lambda_zip                = "lambdas-download/runners.zip"
 
-  enable_organization_runners = false
+  enable_organization_runners = true
```
Comment (ScottGuymer) [inline examples/default/main.tf:48]: More testing im guessing

File: examples/ubuntu/main.tf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 locals {
-  environment = "ubuntu"
+  environment = "niek"
```
Comment (ScottGuymer) [inline examples/ubuntu/main.tf:2]: Testing?

File: examples/default/main.tf (hunk 1 above)
Comment (npalm) [reply examples/default/main.tf:2]: To make me famous :)

**Comment by npalm** [comment]
Tested, latest release downloaded:

image

File: examples/ubuntu/main.tf [hunk 4]
```diff
@@ -93,7 +93,7 @@ module "runners" {
   # Uncomment to enable ephemeral runners
   # delay_webhook_event      = 0
   # enable_ephemeral_runners = true
-  # enabled_userdata         = false
+  # enabled_userdata         = true
```
Comment (npalm) [inline examples/ubuntu/main.tf:96]: Changed example to have the right setting for ephemeral

File: modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/jest.config.js [hunk 5]
```diff
@@ -2,13 +2,13 @@ module.exports = {
...
   collectCoverage: true,
-  collectCoverageFrom: ['src/**/*.{ts,js,jsx}', '!src/**/*local*.ts'],
+  collectCoverageFrom: ['src/**/*.{ts,js,jsx}', '!src/**/*local*.ts', '!src/**/*.d.ts'],
   coverageThreshold: {
     global: {
-      branches: 80,
-      functions: 80,
-      lines: 80,
-      statements: 80
+      branches: 85,
```
Comment (npalm) [inline modules/runner-binaries-syncer/lambdas/runner-binaries-syncer/jest.config.js:8]: Changed settings, overal coverage increased. But som function related to S3 are hard to mock.

File: examples/ubuntu/main.tf (hunk 4 above)
Comment (ScottGuymer) [inline examples/ubuntu/main.tf:96]: Was this changed for testing?

**Review by ScottGuymer — APPROVED** [review]

**Pull request merged**